src/FormLayout/_Abstract.php

```diff
@@ -96,25 +127,27 @@ public function setModel(\atk4\data\Model $model, $fields = null)
         }
 
         if ($fields === null) {
-            $fields = [];
-            foreach ($model->elements as $f) {
-                if (!$f instanceof \atk4\data\Field) {
-                    continue;
-                }
+            $fields = $this->getModelFields($model);
+        }
 
-                if (!$f->isEditable()) {
-                    continue;
-                }
-                $fields[] = $f->short_name;
+        // prepare array of fields - check if fields are editable or read_only/disabled
+        $add_fields = [];
+        foreach ($fields as $field) {
```

Comment:
Checks fields and if field is not editable, then will set disabled=true attribute.
This will work if there is not 2nd argument in setModel() method and also when you pass array of field names. That's for consistency - if field is defined as not editable in model, then there should not be a way to set it as editable in form (you can't save it anyway).